pkg/core/resources/apis/hostnamegenerator/hostname/generator.go

```diff
@@ -134,129 +104,125 @@ func sortGenerators(generators []*hostnamegenerator_api.HostnameGeneratorResourc
 }
 
 func (g *Generator) generateHostnames(ctx context.Context) error {
-	services := &meshservice_api.MeshServiceResourceList{}
-	if err := g.resManager.List(ctx, services); err != nil {
-		return errors.Wrap(err, "could not list MeshServices")
-	}
-
 	generators := &hostnamegenerator_api.HostnameGeneratorResourceList{}
 	if err := g.resManager.List(ctx, generators); err != nil {
 		return errors.Wrap(err, "could not list HostnameGenerators")
 	}
-
 	type serviceKey struct {
 		name string
 		mesh string
 	}
 	type status struct {
 		hostname   string
-		conditions []meshservice_api.Condition
+		conditions []hostnamegenerator_api.Condition
 	}
-	type meshName string
-	type serviceName string
-	type hostname string
-	generatedHostnames := map[meshName]map[hostname]serviceName{}
-	newStatuses := map[serviceKey]map[string]status{}
-	for _, generator := range sortGenerators(generators.Items) {
-		for _, service := range services.Items {
-			serviceKey := serviceKey{
-				name: service.GetMeta().GetName(),
-				mesh: service.GetMeta().GetMesh(),
-			}
-			generatorStatuses, ok := newStatuses[serviceKey]
-			if !ok {
-				generatorStatuses = map[string]status{}
-			}
+	for _, hostnameGenerator := range g.generators {
```

Comment:
Maybe naming here and in the inner loop could be improved, I was confused at first.

Reply:
renamed, let us know if this makes more sense now